runtime/lua/vim/lsp.lua

```diff
@@ -283,7 +283,29 @@ end
 ---
 --- Directory markers (.e.g. '.git/') where the LSP server will base its workspaceFolders,
 --- rootUri, and rootPath on initialization. Unused if `root_dir` is provided.
---- @field root_markers? string[]
+---
+--- It's possibile to specify marker's priority. The default behaviour performs a
+--- directory-up first search, but it can be configured to look for multple markers in each
+--- directory before going up.
+---
+--- Example (default, directory-up first):
+---
+--- ```lua
+---   root_markers = { 'stylua.toml', '.git' }
+--- ```
+---
+--- Looks for `stylua.toml` in every directory going up, if it isn't found it does
+--- the same for `.git`.
+---
+--- Example (mixed priorities):
+---
+--- ```lua
+---   root_markers = { { 'stylua.toml', '.luarc.json' }, '.git' }
+--- ```
+---
+--- Looks for both `.stylua.toml` and `.luarc.json` (in this order) in every
+--- directory going up, if none of them is found it does the same for `.git`.
+--- @field root_markers? (string|string[])[]
```

Comment:
```suggestion
--- Example:
---
--- ```lua
---   root_markers = { 'stylua.toml', '.git' }
--- ```
---
--- Find the first parent directory containing the file `stylua.toml`. If not
--- found, find the first parent directory containing the file or directory
--- `.git`.
---
--- Example:
---
--- ```lua
---   root_markers = { { 'stylua.toml', '.luarc.json' }, '.git' }
--- ```
---
--- Find the first parent directory containing EITHER `stylua.toml` or
--- `.luarc.json`. If not found, find the first parent directory containing the
--- file or directory `.git`.
---
--- @field root_markers? (string|string[])[]
```

Reply:
also helps to put the description in one place, not both above and below the example.

one way to do that is to put the explanation as a comment in the codeblock:

    Example:
    
    ```lua
      -- Find the first ...
      root_markers = ...
    ```

it is disorienting when examples are discussed both above and below each example. important details can be missed by the reader.